DatadogCore/Sources/Core/Storage/Writing/FileWriter.swift

```diff
@@ -37,21 +37,40 @@ internal struct FileWriter: Writer {
     ///  - value: Encodable value to write.
     ///  - metadata: Encodable metadata to write.
     func write<T: Encodable, M: Encodable>(value: T, metadata: M?) {
-        do {
-            var encoded: Data = .init()
-            if let metadata = metadata {
+        var encoded: Data = .init()
+        if let metadata = metadata {
+            do {
                 let encodedMetadata = try encode(value: metadata, blockType: .eventMetadata)
                 encoded.append(encodedMetadata)
+            } catch {
+                DD.logger.error("Failed to encode metadata", error: error)
+                telemetry.error("Failed to encode metadata", error: error)
             }
+        }
 
+        do {
             let encodedValue = try encode(value: value, blockType: .event)
             encoded.append(encodedValue)
+        } catch {
+            DD.logger.error("Failed to encode value", error: error)
+            telemetry.error("Failed to encode value", error: error)
+            return
+        }
 
-            // Make sure both event and event metadata are written to the same file.
-            // This is to avoid a situation where event is written to one file and event metadata to another.
-            // If this happens, the reader will not be able to match event with its metadata.
-            let writeSize = UInt64(encoded.count)
-            let file = try orchestrator.getWritableFile(writeSize: writeSize)
+        // Make sure both event and event metadata are written to the same file.
+        // This is to avoid a situation where event is written to one file and event metadata to another.
+        // If this happens, the reader will not be able to match event with its metadata.
+        let writeSize = UInt64(encoded.count)
+        let file: WritableFile
+        do {
+            file = try orchestrator.getWritableFile(writeSize: writeSize)
+        } catch {
+            DD.logger.error("Failed to get writable file", error: error)
+            telemetry.error("Failed to get writable file", error: error)
```

Comment:
suggestion/ For more context, would it make sense to include `writeSize`? To control the cardinality of `error.message`, this could be done by composing the `error`:
```swift
enum FileWriterError: Error { /* ... */ }

FileWriterError.writeError(error: error, writeSize: writeSize)
```

Reply:
this would mask the underlying error and wrapping one error inside another is something not very encouraged. However I do like the idea of using the writeSize in the error message. Given these are logs, cardinality must not be an issue.